### ISSUE TITLE
JRuby 9.1.8.0builds are failing

### ISSUE THREAD

**Issue by dgmstuart** [varvet/pundit#601]
Builds of JRuby 9.1.8.0 seem to be failing on all branches 

Example failing build:
https://travis-ci.org/varvet/pundit/jobs/567925076

```
$ gem install bundler -v 1.17.3
unsupported Java version "11", defaulting to 1.7
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jruby.ext.openssl.SecurityHelper (file:/home/travis/.rvm/rubies/jruby-9.1.8.0/lib/ruby/gems/shared/gems/jruby-openssl-0.10.2-java/lib/jopenssl.jar) to constructor java.security.cert.CertificateFactory(java.security.cert.CertificateFactorySpi,java.security.Provider,java.lang.String)
WARNING: Please consider reporting this to the maintainers of org.jruby.ext.openssl.SecurityHelper
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
ERROR:  Loading command: install (LoadError)
	library `zlib' could not be loaded: java.lang.NoClassDefFoundError: Could not initialize class org.jruby.ext.zlib.RubyZlib
ERROR:  While executing gem ... (NoMethodError)
    undefined method `invoke_with_build_args' for nil:NilClass
The command "gem install bundler -v 1.17.3" failed and exited with 1 during .
```

There seems to have been a change in the Travis build distribution between the last passing build and this one:

The failing builds use xenial and Ubuntu 16.04
The older passing builds use trusty and Ubuntu 14.04

However, this doesn't seem to be as a result of a change in the Travis configuration.

**Comment by dgmstuart** [comment]
@olleolleolle you've made some commits around JRuby, any ideas? 

I'm wondering if I should specify `dist: trusty`, or if it's worth trying to fix this build on the more recent Ubuntu.

**Comment by dgmstuart** [comment]
Theory: Travis is defaulting to using the latest distribution, and the latest available java version, but JRuby 1 doesn't support Java 11: here's a release note about the JRuby version (2) which starts supporting Java 11: https://www.jruby.org/2018/11/06/jruby-9-2-1-0.html

**Comment by olleolleolle** [comment]
One way is to declare that that build runs on jdk8 versions, like in this PR: https://github.com/cheald/manticore/pull/74

**Comment by dgmstuart** [comment]
@olleolleolle Great tip - thanks

**Comment by dgmstuart** [comment]
Fixed by #600
